src/lexer/mod.rs

```diff
@@ -68,29 +68,45 @@ pub enum TkType {
     Comment,
 }
 
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, Debug)]
 pub struct Location {
     file_name: String,
     line: u32,
     column: u32,
+    pub start: u32,
+    pub end: u32,
```

Comment:
Since `codespan` only accepts byte index than `(line, column)` pair, add these fields to record byte index